packages/engine/src/common/classes/GenerateMeshBVHWorker.ts

```diff
@@ -0,0 +1,94 @@
+import { Box3, BufferGeometry, BufferAttribute, Vector3 } from 'three'
+import { MeshBVH } from 'three-mesh-bvh'
+import { isClient } from '../functions/isClient'
+import Worker from 'web-worker'
+
+export class GenerateMeshBVHWorker {
+  running: boolean
+  worker: Worker
+
+  constructor() {
+    this.running = false
+
+    if (isClient) {
+      // module workers currently don't work in safari and firefox
+      this.worker = new Worker('/workers/generateBVHAsync.worker.js')
+    } else {
+      this._serverWorker()
+    }
+  }
+
+  async _serverWorker() {
+    const path = await require('path')
+    // TODO: Is there a better way to resolve this path?
+    const workerPath = path.resolve(__dirname, '../../../../client/public/workers/generateBVHAsync.worker.js')
```

Comment:
rather than referring to the client dist, we should move this worker file into the engine, as hopefully soon our build chain will allow vite to import workers from within the engine too. so for now, node should reference the worker file in the same folder as this file, and the browser is fine as is for now.